### ISSUE TITLE
New check: NoEmptyLinesBeforeClosingBrace

### ISSUE THREAD

**Issue by mkordas** [checkstyle/checkstyle#2225]
This check would prevent any amount of empty lines before `}`:
```java
void a() {

} // violation above - delete this empty line

void b() {
    if (true) {
    } // OK

} // violation above - delete this empty line
```

This would be effectively the same as prohibiting occurrence of `\s*\n\s*\}` regexp in codebase.

Rationale: Empty lines before closing braces make code longer and they do not improve readability.

**Comment by Skoularikis** [comment]
Hi
I am an undergraduate student and I want to contribute to a famous open-source programm in order to complete a project of mine. It would be also a good opportunity to gain some experience in coding. 
For these reasons, I would be more than happy to give it a try and I am wondering if you can assign me to this issue.

**Comment by rnveach** [comment]
We could have other options for empty lines not wanted.
Like before `case`, `default`, `catch`, or `finally`.
This would make it worth more as a check instead of a rehash of the regexp. `NoEmptyLinesBeforeCheck`.
````
switch (a) {

case 0: // violation above
break;

default: // violation above
break;
}
try {
}

catch (Exception e) { // violation above
}

finally { // violation above
}
````

**Comment by Skoularikis** [comment]
I am willing to give it a try and make a check with name NoEmptyLinesBeforeCheck. Thank you for your help. Can you tell me whether this issue is the same with the existing check EmptyLineSeparatorCheck.java ??
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

**Comment by fprochazka** [comment]
Looks like this is dead. Anybody wants to pick it up? I'm thinking of giving this a try.

**Comment by Vladlis** [comment]
@fprochazka, this issue is not approved. Also I think we had a rule that new checks are only accepted from people with at least 5 contributions, cannot find link to this rule.
@romani, @rnveach, please take a look.

**Comment by fprochazka** [comment]
@Vladlis  

> Also I think we had a rule that new checks are only accepted from people with at least 5 contributions

I'd like to hear the reasoning, it seems unreasonable to me.

**Comment by romani** [comment]
> this issue is not approved. Also I think we had a rule that new checks are only accepted from people with at least 5 contributions, cannot find link to this rule.

it was a rule during moratorium period.
We get out moratorium few month ago, I did a post about this in main-list and removed mentioning of moratorium at main web page.
@fprochazka , please be welcome with PullRequest, we could give you guidance on development process if required, do not hesitate to ask questions.

**Comment by rnveach** [comment]
@romani Didn't this check evolve into requesting empty lines before/after brace in different areas from discussions?
See:
https://groups.google.com/forum/#!topic/checkstyle/rzNJDM_uvyg

**Comment by romani** [comment]
This Check looks like should cover only "no Empty line" aspect, and it could make sense for some body, we will see how it behave on real code during testing
